### ISSUE TITLE
Softmax support for non one-hot targets

### ISSUE THREAD

**Issue by RaffEdwardBAH** [chainer/chainer#729]
According to the [softmax documentation ](http://docs.chainer.org/en/stable/reference/functions.html?highlight=softmax_cross_entropy#chainer.functions.softmax_cross_entropy), the target value 't' must be a one-hot vector of type `int32` . It would be nice if it also supported a target vector type of `float32` for when the target is an actual distribution (ie: the target may have more than one index with a non-zero probability).

**Comment by unnonouno** [comment]
Thank you for your suggestion. Maybe this code works well:
```
loss = -F.sum(F.log(F.softmax(x)) * y)
```
Of course, we can make a new function.
